tools/cldr-code/src/main/java/org/unicode/cldr/test/DisplayAndInputProcessor.java

```diff
@@ -581,9 +588,37 @@ private String inputUnicodeSet(String path, String value) {
         return value;
     }
 
+    /**
+     * Turn all whitespace sequences (including tab and newline, and NBSP for certain paths)
+     * into a single space or a single NBSP depending on path.
+     * Also trim initial/final NBSP, unless the value is only the one character, "\u00A0"
+     *
+     * @param path
+     * @param value
+     * @return the normalized value
+     */
     private String normalizeWhitespace(String path, String value) {
-        // turn all whitespace sequences (including tab and newline, and NBSP for certain paths)
-        // into a single space or a single NBSP depending on path.
+        if (wantsRegularSpace(path)) {
```

Comment:
The cases are, I think:

path doesn't allow NBSP
path doesn't allow SP
path allows either

I think it would be cleaner to have a function that returns an enum {allowSp, allowNbsp, allowSpOrNbsp}

The code for the first two cases looks ok. However, for the third, we want to result in an A0 if there is one in the whitespace sequence, otherwise result in SP. That will need a different REGEX.

Reply:
I see what you mean now about the third case.